app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java

```diff
@@ -423,7 +427,8 @@ public void onInstallComplete(@NonNull String packageName) {
                                     Timber.i("onInstallComplete, package: " + packageName);
                                     holder.btnDownload.setVisibility(View.GONE);
                                     holder.btnInstall.setVisibility(View.GONE);
-                                    executorService.execute(new Runnable() {
+                                    holder.btnUninstall.setVisibility(View.VISIBLE);
+                                    deleteFileExecutor.execute(new Runnable() {
```

Comment:
Is it necessary to delete any _files_ as part an application uninstall functionality? If an application has been installed, any corresponding APK files should have been deleted already, right?

Reply:
I were just replacing `executorService` by `deleteFileExecutor`, since `executorService` is being used for `writeResponseBodyToDisk` task. We should use a dedicated thread for deleting files. Otherwise there'll be problem.